01-js-week-1/setup-linting.md

```diff
@@ -27,75 +33,172 @@ Simply put, style guides help us keep code formatted consistently between develo
-    "commonjs": true,
-    "jquery": true,
-    "jest/globals": true,
-    "es6": true
+    "no-unused-vars": "warn",
+    "no-var": "warn",
+    "no-console": "off",
+    "func-names": "off",
+    "comma-dangle": ["warn", "only-multiline"],
+    "quotes": [
+      "error",
+      "single",
+      { "allowTemplateLiterals": true, "avoidEscape": true }
+    ],
+    "camelcase": "error"
   }
 }
 ```
 
-## Install `Linter-Eslint`
+The file is broken into a few sections:
+
+- **env**:  This defines what kind of environment your JavaScript code will run in.  This is so Eslint knows what to look for.  This example is set to run in the browser and with node (command-line).  It's also set to run in the latest version of Ecmascript (JavaScript).
+- **extends**:  This is a set of pre-defined rules you want to start with.  In this example we're starting with the "standard" eslint rules.  
+  - Later we will use this set to enforce best-practices in React.
+    - `["eslint:recommended", "react-app", "react-app/jest"]` 
+- **parserOptions**: ESLint allows you to specify the JavaScript language options you want to support. By default, ESLint expects ECMAScript 5 syntax. You can override that setting to enable support for other ECMAScript versions as well as JSX by using parser options.
+- **plugins**: These specify additional plugins to eslint to do additional linting.  In this case we're using a module to lint our tests from the Jest testing library.
+- **rules**:  You can then turn individual linting rules on and off.  You can take a look at the [online documentation](https://eslint.org/docs/rules/) for the list of possible rules and what they mean.
+
+You can adjust the `.eslintrc.json` file to add or remove rules you want to use.
 
-Head to VS Code!  Now we can install the extension for eslint and get it set up.
+**Exercise**
+ 
+Take a look at one rule defined in our `.eslintrc.json` file here and look it up in the [Eslint Rules Documentation](https://eslint.org/docs/rules/).  Then answer the following:
 
-In VS Code open the Extensions tab and search for `ESLint`.
+- What does it do?
+- Give an example of a good and bad style under this rule.
+- How do you turn it on or off?
 
-Extensions Tab:
 
-![extensions tab](images/extensions_linter.png)
+## Lets Try!
 
-## Trust But Verify!
+Go ahead and clone [The JS-Calculator Project](https://github.com/ada-c14/js-calculator)
 
-Now open a new .js file and type:
+Then run `npm install` to install the project libraries.
+
+Add the following code to `calculator.js`
 
 ```javascript
-var dont_user_var = "A poorly made variable";
+var i_am_a_variable = true;
+
+console.log("I love Linting!");
 ```
 
-You should get a warning (Unexpected var) about declaring a variable with var, that's ok as we want the warning to verify that linting is working.  
+If you get this, it's working
+
+![linting errors](images/linting.png)
+
+You can click on a warning or error and pick **peek problem** to see what's wrong.
```

Comment:
VS Live Share breaks a lot of these "code lenses", which might be worth mentioning.

Reply:
I didn't know what, mention adding.